rclrs_tests/src/parameter_service_tests.rs

```diff
@@ -0,0 +1,618 @@
+        try_until_timeout(|| {
+            rclrs::spin_once(node.node.clone(), Some(std::time::Duration::ZERO)).ok();
+            rclrs::spin_once(client.clone(), Some(std::time::Duration::ZERO)).ok();
+            *inner_done.read().unwrap()
+        })
+        .await
+        .unwrap();
+    });
+
+    let res = tokio::task::spawn(async move {
+        // Desctibe all parameters
+        let request = DescribeParameters_Request {
+            names: seq![
+                "bool".into(),
+                "ns1.ns2.ns3.int".into(),
+                "read_only".into(),
+                "dynamic".into()
+            ],
+        };
+        let client_finished = Arc::new(RwLock::new(false));
+        let call_done = client_finished.clone();
+        describe_client
+            .async_send_request_with_callback(
+                &request,
+                move |response: DescribeParameters_Response| {
+                    *call_done.write().unwrap() = true;
+                    let desc = response.descriptors;
+                    assert_eq!(desc.len(), 4);
+                    // Descriptors are returned in the requested order
+                    assert_eq!(desc[0].name.to_string(), "bool");
+                    assert_eq!(desc[0].type_, ParameterType::PARAMETER_BOOL);
+                    assert_eq!(desc[0].description.to_string(), "A boolean value");
+                    assert!(!desc[0].read_only);
+                    assert!(!desc[0].dynamic_typing);
+                    assert_eq!(desc[1].name.to_string(), "ns1.ns2.ns3.int");
+                    assert_eq!(desc[1].type_, ParameterType::PARAMETER_INTEGER);
+                    assert_eq!(desc[1].integer_range.len(), 1);
+                    assert_eq!(desc[1].integer_range[0].from_value, 0);
+                    assert_eq!(desc[1].integer_range[0].to_value, 100);
+                    assert_eq!(desc[1].integer_range[0].step, 0);
+                    assert!(!desc[1].read_only);
+                    assert!(!desc[1].dynamic_typing);
+                    assert_eq!(
+                        desc[1].additional_constraints.to_string(),
+                        "Only the answer"
+                    );
+                    assert_eq!(desc[2].name.to_string(), "read_only");
+                    assert_eq!(desc[2].type_, ParameterType::PARAMETER_DOUBLE);
+                    assert!(desc[2].read_only);
+                    assert!(!desc[2].dynamic_typing);
+                    assert_eq!(desc[3].name.to_string(), "dynamic");
+                    assert_eq!(desc[3].type_, ParameterType::PARAMETER_STRING);
+                    assert!(desc[3].dynamic_typing);
+                    assert!(!desc[3].read_only);
+                },
+            )
+            .unwrap();
+        try_until_timeout(|| *client_finished.read().unwrap())
+            .await
+            .unwrap();
+
+        // If a describe parameters request is sent with a non existing parameter, an empty
```

Comment:
As mentioned in earlier review comments, this is not the documented behavior for ROS 2.